src/pages/HomeSelection.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { useEffect, useState } from 'react';
+import firebase from 'firebase';
+import { CircularProgress, makeStyles } from '@material-ui/core';
+import Home from './Home';
+import UserHome from './UserHome';
+
+const useStyles = makeStyles(() => ({
+  progress: {
+    display: 'block',
+    margin: 'auto',
+  },
+}));
+
+export default function HomeSelection() {
+  const classes = useStyles();
+  const [isSignedIn, setIsSignedIn] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [error] = useState();
+
+  function updateUserStatus(user) {
+    setLoading(false);
+    setIsSignedIn(!!user);
+  }
+
+  useEffect(
+    () => firebase.auth().onAuthStateChanged((user) => updateUserStatus(user)),
+    []
+  );
```

Comment:
Here's a simpler way:
```js
  const [isSignedIn, setIsSignedIn] = useState(undefined);
  const loading = isSignedIn == undefined;

  useEffect(
    () => firebase.auth().onAuthStateChanged((user) => setIsSignedIn(!!user)),
    []
  );
```

Reply:
Ok nice that worked but I think that you were missing an `=` so it would be like this `isSiginedIn === undefined`

Reply:
Ah yeah, you're right. I always forget safe equality in JavaScript requires three `=` signs 🤦